hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/util/BackupUtils.java

```diff
@@ -367,9 +367,10 @@ public static String parseHostFromOldLog(Path p) {
       String n = p.getName();
       int idx = n.lastIndexOf(LOGNAME_SEPARATOR);
       String s = URLDecoder.decode(n.substring(0, idx), "UTF8");
-      return ServerName.parseHostname(s) + ":" + ServerName.parsePort(s);
+      final ServerName serverName = ServerName.valueOf(s);
+      return serverName.getHostname() + ":" + serverName.getPort();
```

Comment:
The ServerName class comment says to do sn.getAddress() if you want to do something like what is here. What you reckon? I see you get rid of the getHostAndPort method (good). It too pointed toward sn.getAddress(). What you think?

Its more a comment for going forward. What is here is fine.

Reply:
For `ServerName` I would go with `getHostname` and `getPort` as it is shorter and we internally point to the address. If we want just a single way to get the information we should get rid of the shortcut methods, but I would stick with the shortcuts.

Reply:
Based on your second comment I think I got the point. We could simplify it to just use `getAddress` instead of constructing it ourselves. Let me try that one.